Now `vector_sort.cc` compiles successfully and fresh failing [`test-r-macos-as-cran` log](https://github.com/ursacomputing/crossbow/actions/runs/30450308864/job/90570515768#step:9:2228) points to this remaining newer failure
```console
[ 87%] Building CXX object src/parquet/CMakeFiles/parquet_objlib.dir/arrow/reader.cc.o
/Users/runner/work/crossbow/crossbow/arrow/cpp/src/parquet/arrow/reader.cc:735:46: error: no member named 'ranges' in namespace 'std'
  735 |       const auto first_invalid_offset = std::ranges::adjacent_find(
      |                                         ~~~~~^
/Users/runner/work/crossbow/crossbow/arrow/cpp/src/parquet/arrow/reader.cc:740:26: error: no member named 'ranges' in namespace 'std'
  740 |             start + std::ranges::distance(run_offsets.begin(), first_invalid_offset);
      |                     ~~~~~^
2 errors generated.
make[2]: *** [src/parquet/CMakeFiles/parquet_objlib.dir/arrow/reader.cc.o] Error 1
```

 from yesterday's #50271  `std::ranges::adjacent_find(` and `std::ranges::distance(run_offsets.begin(), first_invalid_offset);`
https://github.com/apache/arrow/pull/50271/changes#diff-6a73dc8ee650cf7b5b8a51f7d4f9f0197823416155c0de7de400294fea588f11R735-R740

--- a/cpp/src/parquet/arrow/reader.cc
+++ b/cpp/src/parquet/arrow/reader.cc
@@ -732,12 +732,12 @@ class PARQUET_NO_EXPORT FixedSizeListReader : public ListReader<int32_t> {
       const int32_t expected_size = has_elements ? list_size : 0;
       std::span<const int32_t> run_offsets(offsets + start,
                                            static_cast<size_t>(length + 1));
-      const auto first_invalid_offset = std::ranges::adjacent_find(
-          run_offsets,
+      const auto first_invalid_offset = std::adjacent_find(
+          run_offsets.begin(), run_offsets.end(),
           [&](int32_t left, int32_t right) { return right - left != expected_size; });
       if (first_invalid_offset != run_offsets.end()) {
         const int64_t x =
-            start + std::ranges::distance(run_offsets.begin(), first_invalid_offset);
+            start + std::distance(run_offsets.begin(), first_invalid_offset);
         const int32_t size = offsets[x + 1] - offsets[x];
         if (has_elements) {
           return Status::Invalid("Expected all lists to be of size=", list_size,

